pilot/pkg/model/virtualservice.go

```diff
@@ -459,7 +460,7 @@ func stringMatchConflict(root, leaf *networking.StringMatch) bool {
 		}
 	}
 
-	return true
+	return false
```

Comment:
Why is this changed?

Reply:
> Why is this changed?

Matchtype is one of exact / prefix / regex. If matchtype is regex, remind users to make sure that there is no conflict between root and leaf，so change to false

Reply:
I believe this changes the behavior when non-regex is used. For example, `prefix: ""` would be `true` before but now `false`?

I would expect the regex section to change only, not the fallback value.